packages/common/src/compiler/filtersCompiler.ts

```diff
@@ -377,30 +378,34 @@ export const renderTableCalculationFilterRuleSql = (
 
     const fieldSql = `${fieldQuoteChar}${getItemId(field)}${fieldQuoteChar}`;
 
+    const escapedFilterRule = escapeStringValuesOnFilterRule(
```

Comment:
extracted escape method into this method , rather than filtering only on string sql, since this can be used in other methods , like numbers (see test)